components/collector/tests/source_collectors/gitlab/test_pipeline_duration.py

```diff
@@ -0,0 +1,57 @@
+"""Unit tests for the GitLab CI-pipeline duration collector."""
+
+from .base import GitLabTestCase
+
+
+class GitLabPipelineDurationTest(GitLabTestCase):
+    """Unit tests for the CI-pipeline duration metric."""
+
+    METRIC_TYPE = "pipeline_duration"
+    METRIC_ADDITION = "min"
+
+    def setUp(self):
+        """Extend to set up fixtures."""
+        super().setUp()
+        self.landing_url = "https://gitlab/project/-/pipelines/1"
+        self.pipeline_json = [
+            {
```

Comment:
Shouldn't we also test that the highest duration is included, when there are multiple pipeline runs?

Reply:
The metric does not measure the slowest pipeline, but the duration of the latest pipeline, so no.